ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/AllocatedParties.scala

```diff
@@ -22,16 +25,35 @@ case class AllocatedParties(
 object AllocatedParties {
   def forExistingParties(parties: List[String]): AllocatedParties = {
     val partiesPrefixMap: Map[String, List[Primitive.Party]] = parties
-      .groupBy(Names.getPartyNamePrefix)
+      .groupBy(Names.parsePartyNamePrefix)
       .view
       .mapValues(_.map(Primitive.Party(_)))
       .toMap
+    val observerPartySetMap = partiesPrefixMap.removedAll(
```

Comment:
Do observer party set parties have a known prefix (say in config).  If so it would seem to be more straightforward to pass this into this method and then process this map in a similar way for all prefixes/groupings.

Reply:
I change the code to look up a partySetPrefix in advance by scanning through stream configs (as this code runs only when there is no submission step)

In general there could be more than one party set but for simplicity I decided to support at most one party set (and failing otherwise)